source/manage/cloud-byok.rst

```diff
@@ -0,0 +1,80 @@
+
+In Mattermost Dedicated, you can use KMS keys in two ways:
+
+- One KMS key for all services
+- Per-service KMS keys (EBS, RDS, S3)
+    - Keys do not need to be unique to each service.
+    - All services must be encrypted at rest.
+    - Selective enablement of this feature can be supported.
+    - Keys do not need to be unique to each service.
+    - In case a customer needs a global database the recommendation from the Infrastructure SRE team is to provide us 2 KMS keys, 1 per region.
+
+Configure BYOK
+------------------------
+1. Customer to provide their AWS KMS ARN to our infrastructure SRE team.
+2. Customer need to add the following blocks to their KMS Policy for the AWS KMS ARN provided
+
+.. code-block:: JSON
+
+    {
+        "Sid": "Allow use of the key",
+        "Effect": "Allow",
+        "Principal": {
+            "AWS": "arn:aws:iam::<MATTERMOST_AWS_ACCOUNT_ID>:user/mattermost-cloud-<environment>-provisioning-<VPC_ID>"
+        },
+        "Action": [
+            "kms:Encrypt",
+            "kms:Decrypt",
+            "kms:ReEncrypt*",
+            "kms:GenerateDataKey*",
+            "kms:DescribeKey"
+        ],
+        "Resource": "<CUSTOM_CUSTOMER_KMS_ID>"
+    },
+    {
+        "Sid": "Allow use of the key role nodes",
+        "Effect": "Allow",
+        "Principal": {
+            "AWS": "arn:aws:iam::<MATTERMOST_AWS_ACCOUNT_ID>:role/nodes.<CLUSTER_ID>-kops.k8s.local"
+        },
+        "Action": [
+            "kms:Encrypt",
+            "kms:Decrypt",
+            "kms:ReEncrypt*",
+            "kms:GenerateDataKey*",
+            "kms:DescribeKey"
+        ],
+        "Resource": "<CUSTOM_CUSTOMER_KMS_ID>"
+    },
+
+3. Infrastructure SRE team need to update kops cluster and S3, RDS resources after KMS policy is updated on customer's end.
+
+An alternative configuration path that Mattermost offers is that the customer provides an external key (non-KMS) to the infrastructure SRE team and the team maintains all the key lifecycle and the relevant resources for the customer.
+This path offers less control to customers but simplifies the setup process.
+
+Requirements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost)
+- Customers oversee the maintenance lifecycle of their custom KMS key.
+- A valid AWS KMS arn for encrypting storage and databases should be provided to the Infrastructure SRE team.
```

Comment:
What is an ``arn``?

Reply:
@cwarnermm fyki: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html#